js/models/messages.js

```diff
@@ -201,21 +201,32 @@
                 this.sendSyncMessage();
             }.bind(this)).catch(function(result) {
                 var now = Date.now();
+                var errors;
+                var conversation = this.getConversation();
                 this.trigger('done');
                 if (result.dataMessage) {
                     this.set({dataMessage: result.dataMessage});
                 }
 
                 if (result instanceof Error) {
-                    this.saveErrors(result);
+                    errors = [result];
                 } else {
-                    this.saveErrors(result.errors);
+                    errors = result.errors;
```

Comment:
This creates a slight change in the order of operations since L215-218 now execute before saveErrors instead of after. Not sure if it matters, but might be best to avoid any unneeded change in behavior.